mllib/src/main/scala/org/apache/spark/ml/classification/GBTClassifier.scala

```diff
@@ -159,14 +157,21 @@ class GBTClassifier @Since("1.4.0") (
     val numFeatures = oldDataset.first().features.size
     val boostingStrategy = super.getOldBoostingStrategy(categoricalFeatures, OldAlgo.Classification)
 
-    val instr = Instrumentation.create(this, oldDataset)
+    val numClasses: Int = getNumClasses(dataset)
```

Comment:
hmm, the logistic regression gets the number of classes and throws in the binomial case, and getNumClasses should ideally get the number of classes from the metadata which shouldn't make an extra pass (ideally the label column is categorical?), but I think it's ok for now to make it 2 until we make GBT support the multiclass case.